docs/design/Refactor Azure authentication.md

```diff
@@ -0,0 +1,165 @@
+  - **Workload Identity**
+  - **Managed Identity**
+  - **Azure CLI**
+- If all attempts fail, the process will return an error.
+
+##### **Proposed Code Snippet**
+```go
+package azure
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+func NewChainedCredential() (*azidentity.ChainedTokenCredential, error) {
+	var creds []azidentity.TokenCredential
+
+	// Add Workload Identity if environment variables are set
+	if tenantID := os.Getenv("AZURE_TENANT_ID"); tenantID != "" {
+		if clientID := os.Getenv("AZURE_CLIENT_ID"); clientID != "" {
+			if tokenFile := os.Getenv("AZURE_FEDERATED_TOKEN_FILE"); tokenFile != "" {
+				wiCred, err := azidentity.NewWorkloadIdentityCredential(&azidentity.WorkloadIdentityCredentialOptions{
+					TenantID:      tenantID,
+					ClientID:      clientID,
+					TokenFilePath: tokenFile,
+				})
+				if err == nil {
+					creds = append(creds, wiCred)
+				}
+			}
+		}
+	}
+
+	// Add Managed Identity
+	if clientID := os.Getenv("AZURE_CLIENT_ID"); clientID != "" {
+		miCred, err := azidentity.NewManagedIdentityCredential(&azidentity.ManagedIdentityCredentialOptions{
+			ID: azidentity.ClientID(clientID),
+		})
+		if err == nil {
+			creds = append(creds, miCred)
+		}
+	}
+
+	// Add Azure CLI Credential
+	cliCred, err := azidentity.NewAzureCLICredential(nil)
+	if err == nil {
+		creds = append(creds, cliCred)
+	}
+
+	if len(creds) == 0 {
+		return nil, fmt.Errorf("no valid credentials detected. Check environment configuration.")
+	}
+
+	// Combine credentials into a chain
+	return azidentity.NewChainedTokenCredential(creds, nil)
+}
+```
+#### **2. Refactor ORAS Auth Providers**
+- Combine `azureidentity.go` and `azureworkloadidentity.go` into a single file.
+- Update the implementation to use the `pkg/common/cloudauthproviders/azure` package for authentication.
+- Authentication type will be inferred based on environment variables.
+
+#### **3. Refactor Key Management and Certificate Providers**
+- Update the providers to leverage the new `pkg/common/cloudauthproviders/azure` package for authentication.
+- Remove redundant logic and ensure consistent authentication processes across all providers.
+
+---
+
+### **Advantages of the Proposed Refactoring**
+
+1. **Improved Maintainability**:
+   - A single SDK (`azidentity`) reduces dependencies and simplifies code management.
+   - Consolidated authentication logic minimizes duplication and enhances clarity.
+
+2. **Enhanced User Experience**:
+   - Automatic detection of authentication type eliminates the need for explicit configuration in most environments.
+
+3. **Extensibility**:
+   - Centralized authentication logic makes it easier to extend support for new scenarios or credential types in the future.
+
+4. **Alignment with Azure Best Practices**:
+   - `azidentity` provides a Kubernetes-native experience, integrating seamlessly with other Azure SDKs.
+
+---
+
+### **Proposed Tasks**
```

Comment:
If we plan to deprecate any existing auth provider/config, we should add this to the V2 tracking issue that @binbin-li made.